test/unit.coffee

```diff
@@ -0,0 +1,382 @@
+assert = require "assert"
+mocha = require "mocha"
+_ = require "underscore"
+thriftPool = {_private} = require "../lib/index"
+async  = require 'async'
+sinon = require "sinon"
+{EventEmitter} = require "events"
+
+connection_mock = ->
+  connection = new EventEmitter()
+  connection.connection =
+    setKeepAlive: sinon.stub()
+  connection.end = sinon.stub()
+  connection
+
+# thrift-pool tests that the exported package properly wraps
+# generic-pool with a passed thrift object, service, and options.
+describe "thrift-pool", ->
+
+  # Initialize mocks to use in thrift-pool creation
+  before ->
+    @generic_error = new Error("error")
+    @mock_connection = connection_mock()
+    @thriftService =
+      Client: class
+        fn: ->
+        fn2: ->
+    @initialized_thrift_service =
+      fn: sinon.stub().yields null, "xyz"
+      fn2: sinon.stub().yields @generic_error, null
+    @thrift =
+      createConnection: =>
+        setImmediate => @mock_connection.emit "connect"
+        @mock_connection
+      createClient: sinon.stub().returns @initialized_thrift_service
+    @wrappedPool =
+      thriftPool @thrift, @thriftService, {"host", "port"}
+
+  afterEach ->
+    @mock_connection.removeAllListeners() # Removes all event listeners
+
+  it "returns an object with the original keys of the thrift service", ->
```

Comment:
supernit: do we really need this test given we have 'returns expected results when called with service functions' below?